### PR TITLE
Search bar

### PR DESCRIPTION
**Pull request by Mallow-Lin**

### CODE REVIEW

File: app/user/[userID]/search/page.tsx [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+  const [selectedDate, setSelectedDate] = useState<ValuePiece|[ValuePiece, ValuePiece]>(new Date());
+
+  const [showCalendar, setShowCalendar] = useState<boolean>(false);
+  const [myLocation, setMyLocation] = useState<LatLngLiteral>();
+
+  const [list, setList] = useState(false);
+  const [searched, setSearched] = useState(false);
+  const [Classes, setClasses] = useState<CLASS[]>([]);
+
+  const apiKey: string | undefined = process.env.GOOGLE_MAPS_API_KEY as string;
```
Comment (kahlstorf1) [inline app/user/[userID]/search/page.tsx]: change to NEXT PUBLIC .env var

File: components/MapView.tsx [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+  time: string,
+  address: string,
+  position: LatLngLiteral,
+}
+
+export default function Map({ center, classes, setList }: MapProps) {
+  const [activeMarker, setActiveMarker] = useState<number | null>(null);
+  const [markers, setMarkers] = useState<marker[]>([]);
+
+  const apiKey: string | undefined = process.env.GOOGLE_MAPS_API_KEY as string;
```
Comment (kahlstorf1) [inline components/MapView.tsx]: NEXT PUBLIC

**Review by kahlstorf1 — COMMENTED** [review]
Other than .env variables looks good to go. Successfully worked locally and npm run build built successfully.

Should be good to merge, but waiting on George/Dannys merge in order to rebase and prevent unnecessary conflicts

**Review by danny-avila — APPROVED** [review]

**Comment by danny-avila** [comment]
lgtm!

**Pull request merged**